staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -46,9 +56,16 @@ func NewHelper(client RESTClient, mapping *meta.RESTMapping) *Helper {
 		Resource:        mapping.Resource.Resource,
 		RESTClient:      client,
 		NamespaceScoped: mapping.Scope.Name() == meta.RESTScopeNameNamespace,
+		GVK:             mapping.GroupVersionKind,
 	}
 }
 
+func (m *Helper) WithDryRun(dryRunServer bool, dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) *Helper {
```

Comment:
I think it's great that this method is optional, so that it doesn't existing compatibility of the helper. It means that you can create that interface in one PR, and then start using it in a separate PR, I think that's the right level of separation. (I wouldn't split for each command).

Now, I'm still not super excited about this interface because you only need `dynamicClient` and `discoveryClient` if dryRunServer is true.